File: Java/boardgames/shared/src/main/java/boardgames/shared/dto/Empty.java
package boardgames.shared.dto;

import java.time.LocalDateTime;
import java.util.List;

// TODO(rune): Er det her Null-Object-Pattern???????????????
public class Empty {
    public static Account account() {
        return new Account(0, "?", "?", "?", "?", Empty.dateTime(), 0, Empty.dateTime());
    }

    public static Match match() {
        return new Match(0, "", 0, 0, Empty.dateTime());
    }

    public static Game game() {
        return new Game(0, "?");
    }

    public static Participant participant() {
        return new Participant(0, 0, 0, 0, Empty.dateTime());
    }

    public static LocalDateTime dateTime() {
        return LocalDateTime.of(1, 1, 1, 1, 1, 1);
    }

    public static <T> List<T> list() {
        return List.of();
    }
}

File: CSharp/Boardgames/Shared/DTO/Empty.cs
namespace GameClient.DTO;

// TODO(rune): Er det her Null-Object-Pattern???????????????
public static class Empty {
    public static Account Account() {
        return new Account {
            AccountId = 0,
            AccountStatus = 0,
            Username = "?",
            FirstName = "?",
            LastName = "?",
            Email = "?",
            CreatedOn = Empty.DateTime(),
            RegisterDateTime = Empty.DateTime(),
        };
    }

    public static Match Match() {
        return new Match {
            MatchId = 0,
            Status = 0,
            State = "?",
            OwnerId = 0,
            GameId = 0,
            CreatedOn = Empty.DateTime(),
        };
    }

    public static Game Game() {
        return new Game {
            GameId = 0,
            Name = "?",
        };
    }

    public static Participant Participant() {
        return new Participant {
            ParticipantId = 0,
            ParticipantStatus = 0,
            MatchId = 0,
            AccountId = 0,
            CreatedOn = Empty.DateTime(),
        };
    }

    public static DateTime DateTime() {
        return new DateTime(1, 1, 1, 1, 1, 1);
    }

    public static List<T> List<T>() {
        return new List<T>();
    }

    public static IEnumerable<T> Enumerable<T>() {
        return System.Linq.Enumerable.Empty<T>();
    }
}
